pkg/steps/lease.go

```diff
@@ -118,8 +118,11 @@ func (s *leaseStep) run(ctx context.Context) error {
 	logrus.Infof("Releasing leases for test %s", s.Name())
 	releaseErr := results.ForReason("releasing_lease").ForError(releaseLeases(client, s.leases))
 
-	// we want a sensible output error for reporting, so we bubble up these individually
-	//if we can, as this is the only step that can have multiple errors
+	return aggregateWrappedErrorAndReleaseError(wrappedErr, releaseErr)
+}
+
+func aggregateWrappedErrorAndReleaseError(wrappedErr, releaseErr error) error {
+	// we want a sensible output error for reporting, so we bubble up these individually if we can
```

Comment:
The code below is useless, utilerrors.NewAggregate [filters out nil errors](https://github.com/openshift/ci-tools/blob/7ae0f5b757d5d9a0bedbcce80ee810460c3ce0bf/vendor/k8s.io/apimachinery/pkg/util/errors/errors.go#L53-L55) so you can just unconditionally do a `return utilerrors.NewAggreate([]error{wrapperdErr, releaseErr}`

Reply:
Not true. We do not want an aggregate, we want the wrapped error, if we can. That's the point of this code.